src/mslice/models/intensity_correction_algs.py

```diff
@@ -220,6 +219,31 @@ def _reduce_bins_along_int_axis(slice_gdos, algorithm, cut_axis, int_axis, cut_a
     return new_ws
 
 
+def _reduce_bins_and_return_signal_error(slice_gdos, algorithm, cut_axis, int_axis, cut_axis_id):
+    ax1 = f'{cut_axis.start}, {cut_axis.step}, {cut_axis.end}'
+    ax2 = f'{int_axis.start}, {int_axis.end - int_axis.start}, {int_axis.end}'
+    if cut_axis_id == 0:
+        ws_out = _cut_nonPSD_general(ax1, ax2, slice_gdos.raw_ws, algorithm)
+    else:
+        ws_out = _cut_nonPSD_general(ax2, ax1, slice_gdos.raw_ws, algorithm)
+    signal = ws_out.extractY()
+    error = ws_out.extractE()
+    return signal, error
+
+
+def _reduce_bins_and_return_signal_error_PSD(slice_gdos, algorithm, int_axis, cut_axis_id, cut_slice_alignment):
+    # PSD type workspace, just sum all the bins
+    signal = np.nansum(slice_gdos.get_signal(), axis=cut_axis_id, keepdims=True)
+    error = np.nansum(np.sqrt(slice_gdos.get_variance()), cut_axis_id, keepdims=True)
```

Comment:
Note that I changed this from `error = np.sqrt(np.nansum(slice_gdos.get_variance()), cut_axis_id, keepdims=True))`

Reply:
@MialLewis sorry I didn't get a chance to check this as I was still at the workshop yesterday but I think this is incorrect. `np.sqrt(slice_gdos.get_variance())` is the same as `slice_gdos.get_error()` but what we need to calculate the error in the sum is to use the quadrature rule:

```
sqrt(e1^2 + e2^2 + e3^3 + ...)
```

(see [wiki](https://en.wikipedia.org/wiki/Propagation_of_uncertainty#Example_formulae)). e1^2 etc are the variances

So it really has to be:

```
error = np.sqrt(np.nansum(slice_gdos.get_variance()), cut_axis_id, keepdims=True))
```

This is the reason the errorbars calculated in the code appears to be very large.

Reply:
No probs... I think the issue was that the code in the test was not changed and was the old code so you reverted the code to make the test pass... Anyway, I think it's ok that this fix goes into nightly.

Thanks for looking into this!